src/fmi_export/CMakeLists.txt

```diff
@@ -1,113 +1,137 @@
-INSTALL(FILES $<TARGET_FILE:fmiGridDynSharedLib> DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime)
-INSTALL(FILES $<TARGET_LINKER_FILE:fmiGridDynSharedLib> DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libraries)
-IF (WIN32)
-	if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win64/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE win64)
-	else("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win32/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE win32)
-	endif("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-ELSE(WIN32)
- IF(UNIX)
-   if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib LIBRARY DESTINATION ${CMAKE_INSTALL_BINDIR}/linux64/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE linux64 )
-	else("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib LIBRARY DESTINATION ${CMAKE_INSTALL_BINDIR}/linux32/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE linux32 )
-	endif("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-	ELSE(UNIX)
-	if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/darwin64/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE darwin64 )
-	else("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-		INSTALL(TARGETS fmiGridDynSharedLib RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/darwin32/ COMPONENT runtime)
-		set(FMILIBRARY_TYPE darwin32 )
-	endif("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-	ENDIF(UNIX)
-ENDIF(WIN32)
-
-target_compile_definitions(fmiBuilder PRIVATE -DGRIDDYNFMILIBRARY_BINARY_LOC="${GRIDDYNFMILIBRARY_BINARY_LOC}")
-target_compile_definitions(fmiBuilder PRIVATE -DFMILIBRARY_TYPE="${FMILIBRARY_TYPE}")
-
-list(APPEND optional_component_test_files ${CMAKE_CURRENT_SOURCE_DIR}/testFMIExport.cpp)
-set (optional_component_test_files ${optional_component_test_files} PARENT_SCOPE)
+install(TARGETS fmiGridDynSharedLib
+        EXPORT griddyn-targets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+# Todo when minimum CMake version becomes 3.6: re-add EXCLUDE_FROM_ALL to these
+# two install FILES commands for fmiGridDynSharedLib
+install(FILES $<TARGET_FILE:fmiGridDynSharedLib>
+        DESTINATION ${CMAKE_INSTALL_BINDIR}
+        COMPONENT
+        RUNTIME)
+install(FILES $<TARGET_LINKER_FILE:fmiGridDynSharedLib>
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT libraries)
+if(WIN32)
+    if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        install(TARGETS fmiGridDynSharedLib
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win64/ COMPONENT
+                RUNTIME)
+        set(FMILIBRARY_TYPE win64)
+    else("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        install(TARGETS fmiGridDynSharedLib
+                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}/win32/ COMPONENT
+                RUNTIME)
```

Comment:
guess we need to rename this component so it isn't thought of as keyword

Reply:
same as https://github.com/LLNL/GridDyn/pull/37#pullrequestreview-143301380